plugins/inputs/procstat/procstat.go

```diff
@@ -354,16 +372,34 @@ func (p *Procstat) getPIDFinder() (PIDFinder, error) {
 }
 
 // Get matching PIDs and their initial tags
-func (p *Procstat) findPids(acc telegraf.Accumulator) ([]PID, map[string]string, error) {
-	var pids []PID
-	tags := make(map[string]string)
-	var err error
+func (p *Procstat) findPids() []PIDS_TAGS_ERR_GROUP {
+	var pids_tags_err_groups []PIDS_TAGS_ERR_GROUP
 
 	f, err := p.getPIDFinder()
 	if err != nil {
-		return nil, nil, err
+		pids_tags_err_groups = append(pids_tags_err_groups, PIDS_TAGS_ERR_GROUP{nil, nil, err})
+	}
+
+	if p.SystemdUnit != "" {
+		groups := p.systemdUnitPIDs()
+		return groups
+	} else if p.CGroup != "" {
+		groups := p.cgroupPIDs()
+		return groups
+	} else {
+		pids, tags, err := p.SimpleFindPids(f)
+		pids_tags_err_groups = append(pids_tags_err_groups, PIDS_TAGS_ERR_GROUP{pids, tags, err})
 	}
 
+	return pids_tags_err_groups
+}
+
+// Get matching PIDs and their initial tags
+func (p *Procstat) SimpleFindPids(f PIDFinder) ([]PID, map[string]string, error) {
```

Comment:
Can you explain why the word "simple" is being used to describe this function ? It seems it could return a `PIDS_TAGS_ERR_GROUP` struct as well.

Reply:
Looking more into this, I see why it is structured this way.